modules/latex.php

```diff
@@ -17,8 +17,160 @@
  *
  * $latex e^{\i \pi} + 1 = 0$  ->  [latex]e^{\i \pi} + 1 = 0[/latex]
  * $latex [a, b]$              ->  [latex][a, b][/latex]
+ *
+ * In addition to supporting old markup, we also now support MathJax math delimiters:
```

Comment:
If we completely disable any MathJax delimiters and only support the existing `$latex ...$` and `[latex]...[/latex]` we won't have any of these issues.

While we're at it we should also replace MathJax's display math delimiters: `$$...$$` and `\[...\]`. Something like `$latex_display ...$` and `[latex_display]...[/latex_display]` would work well and be consistent with existing support.

Here's one we can't get around with new delimiters. MathJax supports these:

```
\begin{equation*}
   e^{\pi i} + 1 = 0
\end{equation*}
```

```
\begin{align}
    f(x) & = Sin(\pi)x - Sin(\pi^2)x + Cos(-\pi^3)*x \\
    \left[\pi\right] & = \left[3.14\right] = 3 \\ 
    \left[\pi ^{2}\right] & = \left[9.8596\right]=9 \\
    \left[-\pi ^{3}\right] & = \left[-9.8596\right]=-10 \\
    f(\frac{\pi }{6}) & = Sin(3\times \frac{\pi }{6})-Sin(9\times \frac{\pi}{6})+Cos(-10\times \frac{\pi }{6})\\
    & = Sin\frac{\pi }{2}-Sin\frac{3\pi }{2}+Cos(\frac{5\pi }{3})\\
    & = 1-\left ( -1 \right )+\frac{1}{2}\\
    & = 2+\frac{1}{2}\\
    & = \frac{5}{2}
\end{align}
```

WP LaTeX explicitly does not support those and it's pretty cool that MathJax does. It has the same potential to break backwards compatibility, but the break may be worth it. I had to at least bring it up since I just discovered it. :)